CMakeLists.txt

```diff
@@ -1,13 +1,21 @@
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.5)
```

Comment:
Split changing the min version and reorganizing into separate PRs. The reorganizing will be fine with a few adjustments. I'm still not clear on the reason for bumping the versions, but not against it.